src/app/admin/pages/edit-book/edit-book.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { BookCategories } from 'src/app/book/constants/books';
+import { BookModel } from 'src/app/core/models/book';
+import { HttpDataService } from 'src/app/core/services/http-data/http-data.service';
+
+@Component({
+  selector: 'app-edit-book',
+  templateUrl: './edit-book.component.html',
+  styleUrls: ['./edit-book.component.scss'],
+})
+export class EditBookComponent implements OnInit {
+  checkoutForm = new FormGroup({});
+  @Input() book!: BookModel;
```

Comment:
inputs should be places first
after them regular inner variables

Reply:
and remove !
throw whole app please

Reply:
fixed